Sources/Fuzzilli/Engines/FuzzEngine.swift

```diff
@@ -20,15 +20,68 @@ public protocol FuzzEngine: ComponentBase {
 }
 
 extension FuzzEngine {
+    private func instrumentForDifferentialTesting(_ program: Program) -> Program {
+        let expectedProbed = Int(fuzzer.config.differentialRate * Double(program.code.count))
+        let b = fuzzer.makeBuilder()
+        var reassignVars: [Variable] = []
+        for _ in 0..<expectedProbed {
+            reassignVars.append(b.loadInt(0))
+        }
+
+        var reassignedVars = 0
+        let differentialHashContext = DifferentialHash().requiredContext
+        b.adopting(from: program) {
+            for instr in program.code {
+                b.adopt(instr)
+
+                guard differentialHashContext.isSubset(of: b.context) else { continue }
+                guard b.numVisibleVariables > reassignVars.count else { continue }
+
+                if probability(fuzzer.config.differentialWeaveRate) {
+                    for _ in 1...10 {
+                        let v = b.randVar()
+                        if !reassignVars.contains(v) {
+                            b.differentialHash(v)
+                            break
+                        }
+                    }
+                }
+
+                guard reassignedVars < reassignVars.count else { continue }
+                if probability(fuzzer.config.differentialRate) {
```

Comment:
IIUC, we could end up not using some of the variables created above?
I'm wondering if we can implement this by setting properties/elements on a global object instead of setting global variables? I.e. at the start we'd create a single object, then here we'd set a property/element on it, and below we'd hash all properties/elements of that object. This way, we don't have to create up-front global variables for the hashes. WDYT?